src/launcher/main.cpp

```diff
@@ -453,15 +485,25 @@ int main(int argc, const char** argv) {
             output = "jfr";
 
         } else if (arg == "--timeout" || arg == "--loop") {
-            params << "," << (arg.str() + 2) << "=" << args.next();
+            const char* value = args.next();
+            params << "," << (arg.str() + 2) << "=" << value;
+            timeout = atoi(value);
             if (action == "collect") action = "start";
 
+        } else if (arg == "--fd-path") {
+            fdtransfer = String(args.next());
+
         } else if (arg == "--fdtransfer") {
             char buf[64];
-            snprintf(buf, sizeof(buf), "@async-profiler-%d-%08x", getpid(), (unsigned int)time_micros());
-            fdtransfer = buf;
+            if (fdtransfer == kEmpty) {
```

Comment:
We don't need this fix because we don't pass `--fdtransfer`, as we're using jattach directly; let's remove? I prefer less diff and only what we truly need.